chain/test_chain.go

```diff
@@ -757,12 +748,20 @@ func (t *TestChain) PendingBlockAddTx(message *core.Message, getTracerFn func(tx
 		ConfigExtensions: t.vmConfigExtensions,
 	}
 
-	// Set the tracer to be used in the vm config
-	tracer := getTracerFn(len(t.pendingBlock.Messages), tx.Hash())
-	if tracer != nil {
-		vmConfig.Tracer = tracer.Hooks
+	// Figure out whether we need to attach any more tracers
+	var extendedTracerRouter *TestChainTracerRouter
+	if len(additionalTracers) > 0 {
+		// If we have more tracers, extend the transaction tracer router's tracers with additional ones
+		extendedTracerRouter = NewTestChainTracerRouter()
+		extendedTracerRouter.AddTracer(t.transactionTracerRouter.NativeTracer())
+		t.transactionTracerRouter.AddTracers(additionalTracers...)
```

Comment:
```suggestion
		// If we have more tracers, extend the transaction tracer router's tracers with additional ones
		extendedTracerRouter = NewTestChainTracerRouter()
		extendedTracerRouter.AddTracer(t.transactionTracerRouter.NativeTracer())
		extendedTracerRouter.AddTracers(additionalTracers...)
```

Reply:
I think we don't want to preserves side-effects by modifying the `transactionTracerRouter`, right?